errors.go

```diff
@@ -297,6 +308,62 @@ func FromGRPCStatus(grpcStatus status.Status) *AsertoError {
 	return result
 }
 
+/**
+ * Retrieves the most inner logger associated with an error.
+ */
+func Logger(err error) *zerolog.Logger {
+	var logger *zerolog.Logger
+
+	if err == nil {
+		return logger
+	}
+
+	for {
+		wErr, ok := err.(*werr.WrappedError)
+		if ok {
+			aErr, aOk := wErr.Err.(*AsertoError)
```

Comment:
You're already handling the `AsertoError` case below. I don't think you need to do it here too.
```go
if wErr, ok := err.(*werr.WrappedError); ok {
    logger = extractLogger(wErr.Ctx)
    err = wErr.Err
}

if aErr, ok := err.(*AsertoError); ok {
    logger = extractLogger(aErr.Ctx)
}

err = erros.Unwrap(err)
...
```

That said, I think you can use `errors.Is()` and/or `errors.As()` instead of doing explicit type checking and unwrapping. I just means that `WrappedError` may need to implement the `Is()` and/or `As()` methods.